neuralet-distancing.py

```diff
@@ -17,28 +18,44 @@ def start_engine(config, video_path):
         logger.warning('Skipping CVEngine as video_path is not set in config file')
 
 
-def start_api(config):
+def start_api(config, message_queue):
     from api.processor_api import ProcessorAPI
-    api = ProcessorAPI(config)
+    api = ProcessorAPI(config, message_queue)
     api.start()
 
+def update_config(process_engine, event, config):
+    logger.info("Restarting CV Engine.")
+    process_engine.terminate()
+    process_engine.join()
+    config.update_config(event['data'], event['options']['save_file'])
+    process_engine = Process(target=start_engine, args=(config,))
+    process_engine.start()
+    logger.info("CV Engine restarted.")
+    return process_engine
 
 def main(config):
     logging.basicConfig(level=logging.INFO)
     if isinstance(config, str):
         config = ConfigEngine(config)
 
-    video_path = config.get_section_dict("App").get("VideoPath", None)
-    process_engine = Process(target=start_engine, args=(config, video_path,))
-    process_api = Process(target=start_api, args=(config,))
+
+    process_engine = Process(target=start_engine, args=(config,))
+    message_queue = Queue()
+    process_api = Process(target=start_api, args=(config, message_queue))
 
     process_api.start()
     process_engine.start()
     logger.info("Services Started.")
 
-    forever = threading.Event()
     try:
-        forever.wait()
+        # Wait for messages
+        while True:
+            event = message_queue.get()
```

Comment:
The docs say that `get()` throws an `Empty` exception when there is no element. have you tested that this is not the case? I think an exception would not be handled correctly here

Reply:
> Queue.get(block=True, timeout=None)
> Remove and return an item from the queue. If optional args block is true and timeout is None (the default), block if necessary until an item is available. If timeout is a positive number, it blocks at most timeout seconds and raises the Empty exception if no item was available within that time. Otherwise (block is false), return an item if one is immediately available, else raise the Empty exception (timeout is ignored in that case).

As I read it if timeout is None (the default in this case) `get` simply waits not raising anything for a message to come by :thinking:

Reply:
OK 👍